### PR TITLE
issue 488 & 489

### PR DESCRIPTION
**Pull request by zhipuch**
Speed ​​up the inference speed of glm-4 validation set and add pure text fine-tuning function for glm-4v

### CODE REVIEW

**Comment by zRzRzRzRzRzRzR** [comment]
A lot of problem need to check again, I add note here
